_posts/2019-02-05-configurable-builds-part-1.md

```diff
@@ -0,0 +1,284 @@
+uses `:android_jdk`. All rules can join in on this. The only obligations rule
+designers have are to write [Starlark rules](https://docs.bazel.build/versions/master/toolchains.html#defining-toolchains)
+describing how their toolchains work and define toolchains for the platforms
+they want to support.
+
+This brings us closer to the goal of `$ bazel build //:all` *just working* by
+replacing ad hoc language-specific flags with a single flag that works everywhere.
+
+### Status
+
+You can use platforms today. The catch is that rules have to opt in support by
+including definitions of how their toolchains work. If you're designing a new
+set of rules, you should design them for platforms. But most existing rules
+predate this work and still rely on legacy flags like
+[`--javabase`](https://source.bazel.build/bazel/+/0b84634f3be1118bdbd501f3d879382d6ae52307:src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java;l=75).
+
+As of this post, Bazel's platform work is heavily focused on rules migration. 
+C++ rules are [near ready](https://github.com/bazelbuild/bazel/issues/6516) and
+due to be [officially integrated](https://github.com/bazelbuild/bazel/issues/7260)
+mid-2019. Java rules will [follow](https://github.com/bazelbuild/bazel/issues/6521) soon after. By year's
+end all major Bazel rules should be integrated. These will be important
+milestones in demonstrating this work's value and providing best practice 
+examples for others.
+
+`--cpu`, `--crosstool_top` and other legacy flags will eventually be removed
+from Bazel. Since rules are migrating on different timelines and projects
+still use these flags for
+[`select`](https://docs.bazel.build/versions/master/configurable-attributes.html
+) and command lines, an [interoperability
+phase](https://github.com/bazelbuild/bazel/issues/6426) will keep them working as long as you need.
+
+For more details, see Bazel's [platforms roadmap](https://bazel.build/roadmaps/platforms.html),
+[configurability roadmap](https://bazel.build/roadmaps/configuration.html),
+[platform](https://docs.bazel.build/versions/master/platforms.html)
+and [toolchain](https://docs.bazel.build/versions/master/toolchains.html)
+docs, and the original [design doc](https://docs.google.com/document/d/1-G-VPRLEj9VyfC6VrQBiR8To-dZjnBSQS66Y4nargGM/edit?ts=57df9619#heading=h.al54v2ddwqzv).
+You can contact the team at [bazel-discuss@googlegroups.com](https://groups.google.com/forum/#!forum/bazel-discuss).
+
+Special thanks to [@katre](https://github.com/katre) for leading Bazel's
+platform vision and [@hlopko](https://github.com/hlopko) and
+[@lberki](https://github.com/lberki) for important C++ contributions.
+
+## Part 2...
+
+Stay tuned for *Configurable Builds - Part 2*, ETA next week. We'll talk about
+building different targets with different settings through *transitions*. We'll
+also discuss why you need to watch your build size when using tlberkhese features.
```

Comment:
Looks like you typo'd lberki's name into "these" :)

Reply:
Fine, we'll make lberki a little bit less famous. :)